addons/purchase_stock/models/purchase.py

```diff
@@ -312,16 +297,41 @@ def _create_or_update_picking(self):
                     line.invoice_lines[0].move_id.activity_schedule(
                         'mail.mail_activity_data_warning',
                         note=_('The quantities on your purchase order indicate less than billed. You should ask for a refund.'))
-
-                # If the user increased quantity of existing line or created a new line
-                pickings = line.order_id.picking_ids.filtered(lambda x: x.state not in ('done', 'cancel') and x.location_dest_id.usage in ('internal', 'transit'))
-                picking = pickings and pickings[0] or False
-                if not picking:
-                    res = line.order_id._prepare_picking()
-                    picking = self.env['stock.picking'].create(res)
-
-                moves = line._create_stock_moves(picking)
-                moves._action_confirm()._action_assign()
+                if float_compare(previous_product_qty[line], line.product_qty, line.product_uom.rounding) >= 0:
+                    if not float_is_zero(line.qty_received, line.product_uom.rounding)\
+                            and float_compare(line.product_qty, line.qty_received, line.product_uom.rounding) > 0:
+                        # is there any quantity to decrease (in a backorder)
+                        open_moves = line.move_ids.filtered(lambda move: move.state not in ('done', 'cancel') and move.location_dest_id.usage != "supplier")
+                        if open_moves and sum(open_moves.mapped('product_uom_qty')) > (line.product_qty - line.qty_received):
```

Comment:
probably an issue with the following example:

- Existing 'open' move with 2 as initial demand
- Reduce the line quantity to have something like 8 as product_qty and 5 as qty receive
You will end with 2 moves having a sum of 5 product_qty